## Final Project Submission

* Student name: Connor Anastasio
* Student pace: self paced (part time)
* Scheduled project review date/time: TBD
* Instructor name: Morgan Jones
* Blog post URL:


In [1]:
# Your code here - remember to use markdown cells for comments as well!

# Aircraft Analysis: Low-risk Aviation Investments

## Overview

This project analyzes the risk, profitability, and opportunity cost of investing in Aviation aircrafts. We will use  of the [Aviation Accident Data](https://www.kaggle.com/datasets/khsamaha/aviation-accident-database-synopses) to create a detailed and informative write-up of which Aircraft are the least likely to be involved in accident. Of those, we will then look at which are the least likely to have fatalies (or alternatively, lowest fatalities).

Descriptive analysis of flight and aircraft data shows that certain types of aircraft are significantly less likely to result in losses than others. We will outline three separate investment strategies and explain the benefit of each.

## Business Problem

![img](./images/animals.png)

The Austin Animal Shelter may be able to improve their resource allocation to both reduce costs and ensure that the center has staff and space to care for the animals brought to them. Doing so will allow the Austin Animal Shelter to better serve its clients while also freeing up resources to expand the scope of services they can offer. Using AAC's animal intake and outcome data, I describe patterns in intakes, stays, and exits to anticipate AAC's outreach, space, and staffing needs for supporting sheltered animals.

In [1]:
#not sure what I'll need so I'll just import everything 
import pandas as pd
import numpy as np
#import seaborn as sns
#import matplotlib.pyplot as plt
#%matplotlib inline
pd.set_option('display.width', None)

We will be using the [Aviation Accident Data](https://www.kaggle.com/datasets/khsamaha/aviation-accident-database-synopses) dataset from Kaggle. It contains very detailed information on aircraft crashes, part failures, and fatalities.

In [5]:
df = pd.read_csv('data/Aviation_Data.csv')

/var/folders/y4/136jbn9s52df1krsdx_9ncrm0000gn/T/ipykernel_3566/3151680963.py:1: DtypeWarning: Columns (6,7,28) have mixed types. Specify dtype option on import or set low_memory=False.
  df = pd.read_csv('data/Aviation_Data.csv')


In [32]:
df = pd.read_csv('data/Aviation_Data.csv', low_memory=False)
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 90348 entries, 0 to 90347
Data columns (total 31 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Event.Id                88889 non-null  object 
 1   Investigation.Type      90348 non-null  object 
 2   Accident.Number         88889 non-null  object 
 3   Event.Date              88889 non-null  object 
 4   Location                88837 non-null  object 
 5   Country                 88663 non-null  object 
 6   Latitude                34382 non-null  object 
 7   Longitude               34373 non-null  object 
 8   Airport.Code            50132 non-null  object 
 9   Airport.Name            52704 non-null  object 
 10  Injury.Severity         87889 non-null  object 
 11  Aircraft.damage         85695 non-null  object 
 12  Aircraft.Category       32287 non-null  object 
 13  Registration.Number     87507 non-null  object 
 14  Make                    88826 non-null

In [33]:
df

Event.Id Investigation.Type Accident.Number  Event.Date  \
0      20001218X45444           Accident      SEA87LA080  1948-10-24   
1      20001218X45447           Accident      LAX94LA336  1962-07-19   
2      20061025X01555           Accident      NYC07LA005  1974-08-30   
3      20001218X45448           Accident      LAX96LA321  1977-06-19   
4      20041105X01764           Accident      CHI79FA064  1979-08-02   
...               ...                ...             ...         ...   
90343  20221227106491           Accident      ERA23LA093  2022-12-26   
90344  20221227106494           Accident      ERA23LA095  2022-12-26   
90345  20221227106497           Accident      WPR23LA075  2022-12-26   
90346  20221227106498           Accident      WPR23LA076  2022-12-26   
90347  20221230106513           Accident      ERA23LA097  2022-12-29   

              Location        Country   Latitude   Longitude Airport.Code  \
0      MOOSE CREEK, ID  United States        NaN         NaN          NaN   
1       BRIDGEPORT, CA  United States        NaN         NaN          NaN   
2        Saltville, VA  United States  36.922223  -81.878056          NaN   
3           EUREKA, CA  United States        NaN         NaN          NaN   
4           Canton, OH  United States        NaN         NaN          NaN   
...                ...            ...        ...         ...          ...   
90343    Annapolis, MD  United States        NaN         NaN          NaN   
90344      Hampton, NH  United States        NaN         NaN          NaN   
90345       Payson, AZ  United States    341525N    1112021W          PAN   
90346       Morgan, UT  United States        NaN         NaN          NaN   
90347       Athens, GA  United States        NaN         NaN          NaN   

      Airport.Name  ... Purpose.of.flight         Air.carrier  \
0              NaN  ...          Personal                 NaN   
1              NaN  ...          Personal                 NaN   
2              NaN  ...          Personal                 NaN   
3              NaN  ...          Personal                 NaN   
4              NaN  ...          Personal                 NaN   
...            ...  ...               ...                 ...   
90343          NaN  ...          Personal                 NaN   
90344          NaN  ...               NaN                 NaN   
90345       PAYSON  ...          Personal                 NaN   
90346          NaN  ...          Personal  MC CESSNA 210N LLC   
90347          NaN  ...          Personal                 NaN   

      Total.Fatal.Injuries Total.Serious.Injuries Total.Minor.Injuries  \
0                      2.0                    0.0                  0.0   
1                      4.0                    0.0                  0.0   
2                      3.0                    NaN                  NaN   
3                      2.0                    0.0                  0.0   
4                      1.0                    2.0                  NaN   
...                    ...                    ...                  ...   
90343                  0.0                    1.0                  0.0   
90344                  0.0                    0.0                  0.0   
90345                  0.0                    0.0                  0.0   
90346                  0.0                    0.0                  0.0   
90347                  0.0                    1.0                  0.0   

      Total.Uninjured Weather.Condition  Broad.phase.of.flight  \
0                 0.0               UNK                 Cruise   
1                 0.0               UNK                Unknown   
2                 NaN               IMC                 Cruise   
3                 0.0               IMC                 Cruise   
4                 0.0               VMC               Approach   
...               ...               ...                    ...   
90343             0.0               NaN                    NaN   
90344             0.0    

We will need to examine columns 6,7 and 28 to determine what format they should be in. It seems there is quite a bit of cleaning to do before we can perform some analysis on this dataset.

In [12]:
seven_col = df.iloc[:, 6] 
eight_col = df.iloc[:, 7]
twonine_col = df.iloc[:, 28]

In [14]:
seven_col.describe()

count       34382
unique      25589
top       332739N
freq           19
Name: Latitude, dtype: object

In [16]:
seven_col.value_counts()

Latitude
332739N      19
335219N      18
334118N      17
32.815556    17
324934N      16
             ..
039613N       1
342034N       1
433113N       1
343255N       1
373829N       1
Name: count, Length: 25589, dtype: int64

In [23]:
eight_col.describe()

count        34373
unique       27154
top       0112457W
freq            24
Name: Longitude, dtype: object

In [24]:
eight_col.value_counts()

Longitude
0112457W       24
1114342W       18
1151140W       17
-104.673056    17
-112.0825      16
               ..
0843135W        1
0101957W        1
1064131W        1
1114414W        1
0121410W        1
Name: count, Length: 27154, dtype: int64

In [21]:
#I don't understand Longitude and Latitude and a lot is missing so let's drop
df.drop(columns = ['Latitude', 'Longitude'], inplace=True)

In [25]:
twonine_col.describe()

count       61724
unique         12
top       Landing
freq        15428
Name: Broad.phase.of.flight, dtype: object

In [27]:
twonine_col.value_counts()

Broad.phase.of.flight
Landing        15428
Takeoff        12493
Cruise         10269
Maneuvering     8144
Approach        6546
Climb           2034
Taxi            1958
Descent         1887
Go-around       1353
Standing         945
Unknown          548
Other            119
Name: count, dtype: int64

In [28]:
twonine_col.isnull().sum()

28624

In [29]:
#drop this as well. It could be interesting, but Unk+other+NaN is twice the size of the biggest category
df.drop(columns = ['Broad.phase.of.flight'], inplace=True)
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 90348 entries, 0 to 90347
Data columns (total 28 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Event.Id                88889 non-null  object 
 1   Investigation.Type      90348 non-null  object 
 2   Accident.Number         88889 non-null  object 
 3   Event.Date              88889 non-null  object 
 4   Location                88837 non-null  object 
 5   Country                 88663 non-null  object 
 6   Airport.Code            50132 non-null  object 
 7   Airport.Name            52704 non-null  object 
 8   Injury.Severity         87889 non-null  object 
 9   Aircraft.damage         85695 non-null  object 
 10  Aircraft.Category       32287 non-null  object 
 11  Registration.Number     87507 non-null  object 
 12  Make                    88826 non-null  object 
 13  Model                   88797 non-null  object 
 14  Amateur.Built           88787 non-null

In [31]:
# Drop Event ID - irrelevant
df['Investigation.Type'].value_counts()

Investigation.Type
Accident      85015
Incident       3874
25-09-2020      702
26-09-2020       60
02-02-2021       39
              ...  
13-09-2021        1
04-08-2021        1
05-08-2022        1
03-11-2020        1
06-01-2021        1
Name: count, Length: 71, dtype: int64

In [ ]:
05-08-2022'

In [ ]:
It'll be much easier to work with this dataset if we normalize column names.

In [5]:
#accident_info = df.copy()
#accident_info.columns = accident_info.columns.str.lower()
#accident_info.columns = accident_info.str.replace('.', '_')
#accident_info.columns = accident_info.strip()
#accident_info

In [11]:
#need to convert object to str first, oops
df

Event.Id Investigation.Type Accident.Number  Event.Date  \
0      20001218X45444           Accident      SEA87LA080  1948-10-24   
1      20001218X45447           Accident      LAX94LA336  1962-07-19   
2      20061025X01555           Accident      NYC07LA005  1974-08-30   
3      20001218X45448           Accident      LAX96LA321  1977-06-19   
4      20041105X01764           Accident      CHI79FA064  1979-08-02   
...               ...                ...             ...         ...   
90343  20221227106491           Accident      ERA23LA093  2022-12-26   
90344  20221227106494           Accident      ERA23LA095  2022-12-26   
90345  20221227106497           Accident      WPR23LA075  2022-12-26   
90346  20221227106498           Accident      WPR23LA076  2022-12-26   
90347  20221230106513           Accident      ERA23LA097  2022-12-29   

              Location        Country   Latitude   Longitude Airport.Code  \
0      MOOSE CREEK, ID  United States        NaN         NaN          NaN   
1       BRIDGEPORT, CA  United States        NaN         NaN          NaN   
2        Saltville, VA  United States  36.922223  -81.878056          NaN   
3           EUREKA, CA  United States        NaN         NaN          NaN   
4           Canton, OH  United States        NaN         NaN          NaN   
...                ...            ...        ...         ...          ...   
90343    Annapolis, MD  United States        NaN         NaN          NaN   
90344      Hampton, NH  United States        NaN         NaN          NaN   
90345       Payson, AZ  United States    341525N    1112021W          PAN   
90346       Morgan, UT  United States        NaN         NaN          NaN   
90347       Athens, GA  United States        NaN         NaN          NaN   

      Airport.Name  ... Purpose.of.flight         Air.carrier  \
0              NaN  ...          Personal                 NaN   
1              NaN  ...          Personal                 NaN   
2              NaN  ...          Personal                 NaN   
3              NaN  ...          Personal                 NaN   
4              NaN  ...          Personal                 NaN   
...            ...  ...               ...                 ...   
90343          NaN  ...          Personal                 NaN   
90344          NaN  ...               NaN                 NaN   
90345       PAYSON  ...          Personal                 NaN   
90346          NaN  ...          Personal  MC CESSNA 210N LLC   
90347          NaN  ...          Personal                 NaN   

      Total.Fatal.Injuries Total.Serious.Injuries Total.Minor.Injuries  \
0                      2.0                    0.0                  0.0   
1                      4.0                    0.0                  0.0   
2                      3.0                    NaN                  NaN   
3                      2.0                    0.0                  0.0   
4                      1.0                    2.0                  NaN   
...                    ...                    ...                  ...   
90343                  0.0                    1.0                  0.0   
90344                  0.0                    0.0                  0.0   
90345                  0.0                    0.0                  0.0   
90346                  0.0                    0.0                  0.0   
90347                  0.0                    1.0                  0.0   

      Total.Uninjured Weather.Condition  Broad.phase.of.flight  \
0                 0.0               UNK                 Cruise   
1                 0.0               UNK                Unknown   
2                 NaN               IMC                 Cruise   
3                 0.0               IMC                 Cruise   
4                 0.0               VMC               Approach   
...               ...               ...                    ...   
90343             0.0               NaN                    NaN   
90344             0.0    

There appear to be redundant columns, columns with indecipherable information, and columns with a majority of null values. Immediately apparent to me is that 'Schedule' only has 12582 values for the 90347 rows, meaning 86% of the data is missing. While itmight contain interesting information, too many values are missing to make it useful.

In [10]:
df = df.drop(df[df['Amateur.Built'] == 'Yes'].index)
df['Amateur.Built'].unique()

array(['No', nan], dtype=object)

In [12]:
missing_values_count = df['Investigation.Type'].isnull().sum()

print(f"Count of missing values in 'Investigation.Type' column: {missing_values_count}")

Count of missing values in 'Investigation.Type' column: 0


In [13]:
df['Investigation.Type'].unique()

array(['Accident', 'Incident', '25-09-2020', '14-06-2021', '03-11-2020',
       '24-02-2021', '02-02-2021', '05-05-2021', '03-12-2020',
       '07-12-2020', '06-05-2021', '19-11-2020', '04-08-2021',
       '13-09-2021', '10-06-2021', '14-12-2021', '08-06-2021',
       '24-06-2021', '24-08-2021', '31-03-2021', '27-05-2021',
       '20-05-2021', '18-11-2020', '25-05-2021', '01-06-2021',
       '15-07-2021', '15-10-2021', '26-09-2020', '03-06-2021',
       '29-09-2021', '28-01-2021', '20-10-2021', '11-12-2020',
       '11-02-2021', '13-12-2021', '06-01-2021', '31-08-2021',
       '04-03-2021', '05-04-2021', '26-01-2021', '29-01-2021',
       '26-08-2021', '08-11-2021', '20-08-2021', '05-01-2021',
       '22-09-2021', '20-09-2021', '16-07-2021', '19-10-2021',
       '16-11-2021', '17-12-2021', '14-06-2022', '22-06-2022',
       '13-10-2022', '03-11-2022', '13-07-2022', '05-07-2022',
       '01-09-2022', '12-08-2022', '08-09-2022', '22-09-2022',
       '01-11-2022', '01-07-2022', '20-07-202

In [20]:
df['Latitude'].isnull().sum()

51672

In [11]:
aviation.drop('Investigation.Type', axis=1, inplace=True)
aviation.drop('Investigation.Type', axis=1, inplace=True)
